### DISCUSSION TITLE
What is the alternative for matchUpdateTypes when using registry URLs?

### DISCUSSION THREAD

**Discussion by ryanpodonnell1** [renovatebot/renovate#26199] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.89.4

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### What is your question?

using the following package manager rule:

```
    {
      matchDatasources: ["docker"],
      registryUrls: ["artifactory.redacted.com:10001"],
      prBodyDefinitions: {
        "Dependency type": "Docker Image"
      },
      matchUpdateTypes: ["minor", "patch", "pin", "digest"],
      automerge: true
    }
```

resulted in the following error:

`ERROR: packageRules[1]: packageRules cannot combine both matchUpdateTypes and registryUrls.`

what would be the proposed alternative?

tried this but doesn't seem like it was catching:

```
{
      matchDatasources: ["artifactory"],
      registryUrls: ["https://artifactory.redacted.com/artifactory/docker-saas/docker-saas/redacted-alpine/"],
      prBodyDefinitions: {
        "Dependency type": "Docker Image"
      },
      matchCurrentVersion: "/^1\\.\\d+\\.\\d+/",
      automerge: true
}
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by viceice** [comment, marked as answer]
split the rules

**Comment by ryanpodonnell1** [comment]
> split the rules 

Oh is packageRules just like a pipeline? Where I can pipe outputs from one rule down the list and catch with another rule?

I just thought that the last rule matched took precedence and that was it like an all or nothing.

**Reply by rarkins** [reply]
Update types are known after versions lookup. Registry URLs are applied before lookup. So it's impossible to apply a registry url based on an update type - your existing rule will be pointless because it's applied too late
